src/App.tsx

```diff
@@ -1,26 +1,186 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import React, { useEffect, useRef, useState } from 'react';
+import { getTodos, postTodo, removeTodo } from './api/todos';
+import { Todo } from './types/Todo';
+import { TodosList } from './components/TodosList';
+import { FilterType } from './types/FilterType';
+import { Footer } from './components/Footer';
+import { Header } from './components/Header';
+import { ErrorType } from './types/ErrorType';
+import { ErrorNotification } from './components/ErrorNotification';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [error, setError] = useState<ErrorType | null>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState<FilterType>(FilterType.All);
+  const [tmpTodo, setTmpTodo] = useState<Omit<Todo, 'userId'> | null>(null);
+  const [updatedTodosId, setUpdatedTodosId] = useState<number[]>([]);
+
+  const downloadTodos = async () => {
+    try {
+      const tmp = await getTodos();
+
+      setTodos(tmp);
+      setFilteredTodos(tmp);
+    } catch {
+      setError(ErrorType.UnableToLoad);
+    }
+  };
+
+  useEffect(() => {
+    downloadTodos();
+    inputRef.current?.focus();
+  }, []);
+
+  useEffect(() => {
+    if (todos) {
+      switch (query) {
+        case FilterType.Active:
+          setFilteredTodos(todos.filter(todo => !todo.completed));
+          break;
+        case FilterType.Complited:
+          setFilteredTodos(todos.filter(todo => todo.completed));
+          break;
+        default:
+          setFilteredTodos(todos);
+          break;
+      }
+    }
+  }, [todos, query]);
+
+  const returnError = () => {
+    switch (error) {
+      case ErrorType.EmptyTitle:
+        return 'Title should not be empty';
+      case ErrorType.UnableToDelete:
+        return 'Unable to delete a todo';
+      case ErrorType.UnableToLoad:
+        return 'Unable to load todos';
+      case ErrorType.UnableToUpdate:
+        return 'Unable to update a todo';
+      case ErrorType.UnableToAdd:
+        return 'Unable to add a todo';
+      default:
+        return null;
+    }
+  };
```

Comment:
This can be colocated down to component that shows error as you already pass error with props